### ISSUE TITLE
Lambda layer rename

### ISSUE THREAD

**Issue by sun-peach** [keras-team/keras#13029]
**System information**  
- Have I written custom code (as opposed to using example directory):   No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu16.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13
- Keras version:  latest
- Python version:  3.5+
- CUDA/cuDNN version: 10.0/7.5 
- GPU model and memory:  2080 ti

I define the Lambda layer as below:
    output = Lambda(lambda x: K.concatenate(x, axis=1),name="Output)(all_outputs)

I expect the output.name is "Output", however it is "Output/concat:0". Since I use dict format for passing data, the name really matters a lot. How can I get rid of the "/concat" part?

BTW, a further question: this is the output of my sub model "model_2", so the final name of this layer is "model_2/Output/concat:0". If I want to pass data in dict format (say the data is y), should I do {"model_2/Output/concat": y} ?

Thank you very much!
